### PR TITLE
Enable recursive components

### PR DESCRIPTION
**Pull request by theotherjimmy**
### Description

Should unblock the recursive `components_add` you're doing

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

---
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/88)

### CODE REVIEW

File: tools/config/__init__.py [hunk 1]
```diff
@@ -1245,6 +1245,7 @@ def load_resources(self, resources):
         """
         # Update configuration files until added features creates no changes
         prev_features = set()
+        prev_components = set()
```
Comment (alzix) [inline tools/config/__init__.py]: labels may also modify scanning rules

Comment (theotherjimmy) [reply tools/config/__init__.py]: That is a solid point @alzix. I'll get to that today.

Comment (theotherjimmy) [reply tools/config/__init__.py]: It's almost like I need to be comparing all cumulative attributes...

Comment (theotherjimmy) [reply tools/config/__init__.py]: Turns out that this is not possible:  `Target override 'target.extra_labels' in library:alpha[*] is only allowed at the application level` is the error you get when you try.

**Comment by alzix** [comment]
Good to know.

File: tools/config/__init__.py (hunk 1 above)
Comment (alzix) [reply tools/config/__init__.py]: Good to know

**Pull request merged**